packages/dht/src/dht/DhtNode.ts

```diff
@@ -112,24 +117,62 @@ const logger = new Logger(module)
 
 export type Events = TransportEvents & DhtNodeEvents
 
-export const createPeerDescriptor = (msg?: ConnectivityResponse, nodeId?: DhtAddress): PeerDescriptor => {
+const calculateNodeIdRaw = (signingModule: ISigningModule, ipAddress: number,
+    salt: Uint8Array): DhtAddressRaw => {
+    // nodeId is calculated as 
+    // concatenate(get104leastSignificatBits(hash(ipAddress)), 
+    //  get56leastSignificatBits(sign(concatenate(ipAddress, salt))))
+
+    const ipAsBuffer = Buffer.alloc(4)
+    ipAsBuffer.writeUInt32BE(ipAddress)
+
+    const ipHash = signingModule.hash(ipAsBuffer)
+
+    const idPartToSign = Buffer.concat([ipAsBuffer, salt])
+    const signature = signingModule.sign(idPartToSign)
+
+    const nodeIdRaw = Buffer.concat([ipHash.slice(ipHash.length - 13, ipHash.length),
+        signature.slice(signature.length - 7, signature.length)])
+
+    return nodeIdRaw
+}
+
+export const createPeerDescriptor = (signingModule: ISigningModule, 
+    msg?: ConnectivityResponse, nodeId?: DhtAddress, nodeName?: string): PeerDescriptor => {
```

Comment:
We won't support that users could give explicit nodeId, i.e this parameter and all usages to this can be removed.

Reply:
Support for now.